upgradability/set_code_hash/voting_fixed/README.md

```diff
@@ -0,0 +1,69 @@
+## Voting fixed example
+
+This smart contract is part of an example showing how we can write upgradable smart contracts in ink! using `set_code_hash`.
+
+See `voting` and `transparent_voting` examples first.
+
+### Description
+
+This contract provides a simple voting interface which allows voting for one of some two options. However, when compared to
+`voting`, this contract has ability to store more votes and forbids adding new voters or
+upgrading itself.
+
+The goal of this example is to show how one might perform a smart contract upgrade (using `set_code_hash` method) in a case in which
+storage layouts of the original contract and the new one differ.
+
+### Upgrading `voting` smart contract: upgrade with storage layout change
+
+For an introduction of how to upgrade with `set_code_hash` see `transparent_voting` example.
+
+#### Example scenario
+
+Let's say you deployed `voting` contract, updated list of the allowed voters to it's
+desired state and gathered some votes. Then you realized that the contract has a flaw:
+votes are stored in `u8` counters, and will soon overflow.
+
+Now, you want to fix this issue changing the counters to `u64`, and remove ability
+to add new voters/upgrade the contract (to assure users that the list of voters is final).
+
+#### Storage collisions
+
+The big difference between the upgrade in `transparent_voting` example, and in this one, is that here we want to change the storage layout.
+We cannot simply change counters' type to `u64`, we would not be able to read any of our
+old data, this time we need to store the new data in a different place. One way of doing it is to
+append new fields at the end of our storage struct, another is to keep the new data under
+a different storage key.

```

Comment:
you seem to be doing both things - what am I missing?

Reply:
hmm, now I understand that you do not "keep new data under a different storage key", but instead you use openbrush macro - can you please explain this "keeping new data under a different storage key"?

Reply:
OK, as far as I understand, every field of `Storage` is kept at a different storage key, which is like a key for a map/Mapping.
If we do not specify at what key we want out field to be stored, then they are assigned some default keys. Just appending new field gives it some default key. However, we can specify a specific storage key for part of the storage, and that is one of the things that this macro does.